### PR TITLE
Refactor of keywords / reserved names page

### PR DESCRIPTION
**Pull request by AndrewDiMola**
Refactor of keywords page to be more exhaustive and discourage other built-in types and literals when naming things.  Not sure if there warrants separation between keywords and "contextual keywords"

image

### CODE REVIEW

**Comment by AndrewDiMola** [comment]
ref: #1083

**Comment by lexidor** [comment]
Nothing with your refactoring perse, but something that jumped out to me. This list of keywords contains things that have long since been removed from the language, like `and`, `or`, `xor`, `instanceof`. It may be worth looking at them to see if their special behavior as an operator has been completely removed. For example, the following code typechecks and runs without issue.

```HACK
const int and = 6;

function and(bool $a, bool $b): bool {
  return $a && $b;
}

final class and {
  public function and(): int {
    return 7;
  }

  public static function create(): and {
    return new and();
  }
}

<<__EntryPoint>>
function main(): void {
  $and = and::create();
  if (and(true, and + $and->and() === 13)) {
    echo "Is `and` a keyword?\n";
  }
}

```

**Comment by lexidor** [comment]
`null` is missing from the list of built-in types.

**Comment by AndrewDiMola** [comment]
I ended up making a section for deprecated keywords and removing the types, opting to link to the full section instead

image

File: test.hack [hunk 1]
```diff
@@ -0,0 +1,13 @@
+namespace Hack\GettingStarted\MyFirstProgram;
+namespace HH\Lib\Str;
```
Comment (lexidor) [inline test.hack:2]: This file is declared in two namespaces. The MyFirstProgram and Str namespace.

Comment (fredemmott) [inline test.hack:2]: This should be `use namespace HH\Lib\Str;` ...

File: test.hack [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+<<__EntryPoint>>
+function main(): void {
```
Comment (fredemmott) [inline test.hack:5]: This currently defines `HH\Lib\Str\main` which is unwanted

File: test.hack [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  $x = "Apples Oranges and ";
+  $y = "Bananas";
+  splice($x, $y, 1);
```
Comment (fredemmott) [inline test.hack:9]: ... and this should be `Str\splice`

**Comment by AndrewDiMola** [comment]
Oh geez @fredemmott @lexidor did not mean to add that file in this commit

File: test.hack (hunk 1 above)
Comment (AndrewDiMola) [reply test.hack]: good feedback! was just messing around and accidentally added when I wasn't paying attention -- removed.

File: test.hack (hunk 3 above)
Comment (AndrewDiMola) [reply test.hack]: good feedback! was just messing around and accidentally added when I wasn't paying attention -- removed.

File: test.hack (hunk 2 above)
Comment (AndrewDiMola) [reply test.hack]: good feedback! was just messing around and accidentally added when I wasn't paying attention -- removed.

**Comment by AndrewDiMola** [comment]
dangers of `git add .` and working too late at night

**Review by fredemmott — APPROVED** [review]

**Pull request merged**